tests/unit/user/payment.test.ts

```diff
@@ -30,22 +35,274 @@ describe ("Payment Tests", ()=>{
         await userController.createUser({
             ...userInfo
         })
+        let userId;
+        try {
+            const user = await User.findOne({
+                email: "michaeltest@gmail.com"
+            }).exec();
+            
+            if (user) {
+                userId = user._id;
+            } else {
+                console.log("No user found with that email");
+            }
+        } catch (err) {
+            console.error(err);
+        }
+     
+        let date = new Date();
+        const paymentInfo = {
+            memberId: userId?.valueOf(),
+            paymentAmount: 20,
+            paymentDate: date,
+            paymentStatus: Status[0].toString()
+        };
+        
+        
+        await paymentController.createPayment({ 
+            ...paymentInfo
+        })
+
+        const paymentInstance = await Payments.findOne({
+            memberId: userId?.valueOf()
+        }).exec()
+        
+        expect(paymentInstance).to.exist;
+        expect(paymentInstance?.paymentDate).to.include(paymentInfo.paymentDate);
+        expect(paymentInstance?.memberId).to.include(paymentInfo.memberId);
+        expect(paymentInstance?.paymentAmount).to.equal(paymentInfo.paymentAmount);
+        expect(paymentInstance?.paymentStatus).to.equal(paymentInfo.paymentStatus);
+        
+        
+})  
+    it ("PaymentController::findUserPaymentHistory finds user and sort them in order by date", async () =>{
+       //creating two different users
+       await Payments.deleteMany({});
+       await Payments.deleteMany({});
+        const userInfo1 = {
+            firstName: "mahkel",
+            lastName: "test",
+            email: "michaeltest@gmail.com",
+            password: "test"
+        };
+        await userController.createUser({
+            ...userInfo1
+        })
+        
+        let userId: Object|undefined;
+        try {
+            const user = await User.findOne({
+                email: "michaeltest@gmail.com"
+            }).exec();
+            
+            if (user) {
+                userId = user._id;
+            } else {
+                console.log("No user found with that email");
+            }
+        } catch (err) {
+            console.error(err);
+        }
+        
+        for (let i = 0; i < 3 ; i++){
+            
+            const paymentInfo = {
+                memberId: userId?.valueOf(),
+                paymentAmount: Math.floor(Math.random() * (69-3)+3),
+                paymentDate: new Date(),
```

Comment:
What if you create 3 diffferent dates beforehand (separate them apart by like a week, for example), and use those instead? I'm thinking back to that error we had with dates being equal (despite the tests failing), so it'd be safe to separate the dates out manually

Reply:
noted.